src/common/transformer_ctx.h

```diff
@@ -51,6 +74,9 @@ struct DecoderContext {
     // norm epsilon
     float epsilon;
 
+    // rope scaling parameters
+    RopeParams ropeParams;
```

Comment:
Is it a must to put RopeParams here? As DecoderContext is used again and again, keep it small may have the potential performance benefit.
If really needed in DecoderContext, suggest to make it as a pointer. so that we could make sure less impact to the models not using this param.

Reply:
There are several variants of finetuned llama2 rope embeddings, all of them require parameters to be passed in. Tried some way and still put the rope params in ctx, which has been changed to a pointer.